pkg/plan_modifier/pg_bouncer.go

```diff
@@ -39,51 +39,64 @@ func (m CustomPgBouncerModifier) PlanModifyObject(ctx context.Context, req planm
 		return
 	}
 
-	// have plan settings combine with state settings if planned settings len > 0
-	if !req.PlanValue.IsNull() && len(req.PlanValue.Attributes()["settings"].(basetypes.SetValue).Elements()) > 0 {
-		if !req.PlanValue.Attributes()["is_enabled"].(basetypes.BoolValue).ValueBool() {
-			if !req.PlanValue.Attributes()["settings"].(basetypes.SetValue).IsNull() {
-				resp.Diagnostics.AddError("if pg_bouncer.is_enabled = false then pg_bouncer.settings should be removed", "please remove pg_bouncer.settings if pg_bouncer.is_enabled = false")
-				return
-			}
-		} else if req.PlanValue.Attributes()["is_enabled"].(basetypes.BoolValue).ValueBool() {
-			newPlanWithPrefilledPlannedSettings := resp.PlanValue.Attributes()["settings"].(basetypes.SetValue).Elements()
-			stateSettings := []attr.Value{}
-			if len(req.StateValue.Attributes()) != 0 {
-				stateSettings = req.StateValue.Attributes()["settings"].(basetypes.SetValue).Elements()
-			}
+	if !req.PlanValue.IsNull() {
+		reqPlanSettings := req.PlanValue.Attributes()["settings"].(basetypes.SetValue)
+		// have plan settings combine with state settings if planned settings len > 0
+		if len(reqPlanSettings.Elements()) > 0 {
+			reqPlanIsEnabled := req.PlanValue.Attributes()["is_enabled"].(basetypes.BoolValue)
+			if !reqPlanIsEnabled.ValueBool() {
+				if !req.PlanValue.Attributes()["settings"].(basetypes.SetValue).IsNull() {
+					resp.Diagnostics.AddError("if pg_bouncer.is_enabled = false then pg_bouncer.settings should be removed", "please remove pg_bouncer.settings if pg_bouncer.is_enabled = false")
```

Comment:
Would it handle the `settings = []` case?

Reply:
I think settings = [] is already handled below if enabled is true. It doesn't allow it. But I put in another check so if enabled is false it  shouldn't allow settings = [] also